includes/classes/class-fixinsecurecontent.php

```diff
@@ -0,0 +1,343 @@
+	 * @var array
+	 */
+	private array $warning_count = array();
+
+	/**
+	 * Class instance
+	 *
+	 * @var ?FixInsecureContent
+	 */
+	protected static ?FixInsecureContent $instance = null;
+
+	/**
+	 * Get class instance
+	 *
+	 * @return FixInsecureContent
+	 */
+	public static function get_instance() {
+		if ( null === self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Gets the "post_type" args for the WP_Query calls for the functionality to fix insecure content from within the admin UI.
+	 *
+	 * @return array
+	 */
+	public function get_wp_query_post_type_args(): array {
+		return array(
+			'show_ui'      => true,
+			'public'       => true,
+			'show_in_rest' => true,
+		);
+	}
+
+	/**
+	 * Counts posts to be checked for insecure content.
+	 *
+	 * @param string $post_type      The post type.
+	 * @param int    $posts_per_page The batch size.
+	 * @param int    $post_offset    The post offset.
+	 *
+	 * @return int
+	 */
+	public function count_post_to_check( string $post_type, int $posts_per_page, int $post_offset ): int {
+		// Exclude post from post types not shown in REST when counting insecure content to fix from the admin UI.
+		$post_types = get_post_types( $this->get_wp_query_post_type_args() );
+
+		$args = array(
+			'posts_per_page' => $posts_per_page,
+			'post_type'      => 'all' === $post_type ? $post_types : $post_type,
+			'offset'         => $post_offset,
+			'nopaging'       => true,
+		);
+
+		$query = new WP_Query( $args );
+
+		return $query->found_posts;
+	}
+
+	/**
+	 * Runs the fixing procedure for insecure content.
+	 *
+	 * @param bool|string $include        Comma separated IDs of post if fixing post content for multiple posts.
+	 * @param bool        $all            Flag for running the fix on all posts.
+	 * @param string      $post_type      The post type.
+	 * @param int         $posts_per_page The batch size.
+	 * @param int         $post_offset    The post offset.
+	 * @param bool        $dry_run        Flag for dry or normal run.
+	 * @param array       $args           WP CLI arguments.
+	 *
+	 * @return array|void
+	 */
+	public function fix( $include, $all, $post_type, $posts_per_page, $post_offset, $dry_run, $args = array() ) {
+		$this->dry_run = $dry_run;
+
+		if ( defined( 'WP_CLI' ) && WP_CLI ) {
+			WP_CLI::log( __( 'Checking post content...', 'insecure-content-warning' ) );
+		}
+
+		if ( false === $all && ! empty( $args[0] ) ) {
+			$this->fix_insecure_content( $args[0] );
+		} elseif ( false === $all && empty( $args ) && false !== $include ) {
+			$post_ids_list = explode( ',', $include );
+			foreach ( $post_ids_list as $icw_post_id ) {
+				$this->fix_insecure_content( trim( $icw_post_id ) );
+			}
+			$this->total_post_count = count( $post_ids_list );
+		} else {
+			$total = 0;
+
+			// Exclude post from post types not shown in REST when fixing insecure content from the admin UI.
+			if ( ! defined( 'WP_CLI' ) && 'any' === $post_type ) {
```

Comment:
you've used `all` elsewhere and `any` here.

Reply:
Good point, fixed.